Weather-Monitoring-And-Reporting-Service_Tests/PublisherTest/WeatherPublisherTest.cs

```diff
@@ -0,0 +1,122 @@
+﻿using AutoFixture.AutoMoq;
+using AutoFixture;
+using Moq;
+using Weather_Monitoring_And_Reporting_Service;
+using Weather_Monitoring_And_Reporting_Service.Configuration;
+using Weather_Monitoring_And_Reporting_Service.Publisher;
+using Weather_Monitoring_And_Reporting_Service.Strategies;
+using Weather_Monitoring_And_Reporting_Service.Subscriber;
+using Xunit;
+using Weather_Monitoring_And_Reporting_Service.WeatherBot;
+
+namespace Weather_Monitoring_And_Reporting_Service_Tests.PublisherTest;
+
+public class WeatherPublisherTest
+{
+    private readonly IFixture fixture;
+
+    public WeatherPublisherTest()
+    {
+        fixture = new Fixture().Customize(new AutoMoqCustomization());
+    }
+
+    private WeatherPublisher CreateWeatherDataPublisher()
+    {
+        var mockTextFormatStrategy = new Mock<ITextFormatStrategy>();
+        var text = fixture.Create<string>();
+        var botConfig = fixture.Create<BotConfiguration>();
+
+        mockTextFormatStrategy.Setup(x => x.GetWeatherData(It.IsAny<string>())).Returns(fixture.Create<Weather>());
+
+        return new WeatherPublisher(text, mockTextFormatStrategy.Object, botConfig);
+    }
+
+
+    [Fact]
+    public void WeatherDataPublisher_AttachSubscriber()
```

Comment:
where is the expected?

Reply:
?

Reply:
That's Right ?

[Fact]
public void WeatherDataPublisher_AttachSubscriber()
{
    var sut = CreateWeatherDataPublisher();
    var mockSubscriber = _fixture.Create<Mock<IWeatherSubscriber>>();

    sut.Attach(mockSubscriber.Object);

    Assert.Contains(mockSubscriber.Object, sut.Subscribers);

    Assert.True(sut.Subscribers.Contains(mockSubscriber.Object));
}

Reply:
I mean something like `WeatherDataPublisher_AttachSubscriber_ShouldWork`